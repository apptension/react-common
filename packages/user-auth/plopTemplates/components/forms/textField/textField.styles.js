import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
`;

export const InputContainer = styled.div`
  width: 100%;
  display: flex;
`;

export const Label = styled.label`
  width: 30%;
`;

export const Input = styled.input`
  width: 70%;
`;

export const Error = styled.div`
  width: 100%;
  color: red;
  padding-left: 30%;
`;
