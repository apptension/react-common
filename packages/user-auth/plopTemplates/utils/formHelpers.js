import reportError from 'report-error';
import { List } from "immutable";
import { when, is, pipe } from 'ramda';

export const getFormFieldHelperText = (intl, messages, { error, invalid, touched }) => {
  if (invalid && touched) {
    try {
      return intl.formatMessage(messages[pipe(
        when(
          is(List),
          (errors) => errors.first()
        ),
        when(
          is(Array),
          ([firstError]) => firstError
        ),
      )(error)]);
    } catch (e) {
      reportError(e);
    }
  }
  return '';
};
