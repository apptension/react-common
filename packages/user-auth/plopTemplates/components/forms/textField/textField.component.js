import React, { PureComponent } from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import { renderWhenTrue } from '../../../utils/rendering';
import { getFormFieldHelperText } from '../../../utils/formHelpers';
import { Container, InputContainer, Label, Input, Error } from './textField.styles';

export class TextField extends PureComponent {
  static propTypes = {
    input: PropTypes.object.isRequired,
    meta: PropTypes.object.isRequired,
    label: PropTypes.object.isRequired,
    type: PropTypes.string.isRequired,
    messages: PropTypes.object.isRequired,
    intl: PropTypes.object.isRequired,
  };

  renderError = renderWhenTrue(() => {
    const { meta, intl, messages } = this.props;
    return <Error>{getFormFieldHelperText(intl, messages, meta)}</Error>;
  });

  render() {
    const { label, input, type, meta } = this.props;
    return (
      <Container>
        <InputContainer>
          <Label>
            <FormattedMessage {...label} />
          </Label>
          <Input {...input} type={type} />
        </InputContainer>
        {this.renderError(meta.touched && !!meta.error)}
      </Container>
    );
  }
}
