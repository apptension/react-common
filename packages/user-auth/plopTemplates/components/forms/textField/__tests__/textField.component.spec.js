import React from 'react';
import { shallow } from 'enzyme';

import { TextField } from '../textField.component';
import { intlMock } from '../../../../utils/testUtils';

describe('<TextField />', () => {
  const defaultProps = {
    intl: intlMock(),
    input: {},
    meta: {
      touched: false,
      error: null,
      invalid: null,
    },
    label: {
      id: 'text',
    },
    type: 'text',
    messages: {
      error: {
        id: 'test',
        defaultMessage: 'Text error',
      },
    },
  };

  const component = (props) => shallow(
    <TextField {...defaultProps} {...props} />
  );

  it('should render itself', () => {
    const wrapper = component();
    global.expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render error state', () => {
    const wrapper = component({
      meta: {
        touched: true,
        error: 'error',
        invalid: true,
      },
    });
    global.expect(wrapper.dive()).toMatchSnapshot();
  });
});
