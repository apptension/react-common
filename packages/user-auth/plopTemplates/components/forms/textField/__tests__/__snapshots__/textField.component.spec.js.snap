// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render error state 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 10px;
}

<div
  className="c0"
>
  <styled.div>
    <styled.label>
      <FormattedMessage
        id="text"
        values={Object {}}
      />
    </styled.label>
    <styled.input
      type="text"
    />
  </styled.div>
  <styled.div>
    test / Text error / {}
  </styled.div>
</div>
`;

exports[`<TextField /> should render itself 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 10px;
}

<div
  className="c0"
>
  <styled.div>
    <styled.label>
      <FormattedMessage
        id="text"
        values={Object {}}
      />
    </styled.label>
    <styled.input
      type="text"
    />
  </styled.div>
</div>
`;
